ansys/mapdl/core/licensing.py

```diff
@@ -0,0 +1,456 @@
+"""Module for licensing and license serve checks."""
+
+import logging
+import os
+import time
+import re
+import socket
+import subprocess
+
+
+from ansys.mapdl.core.errors import LicenseServerConnectionError
+from ansys.mapdl.core.misc import threaded
+
+LOCALHOST = "127.0.0.1"
+LIC_PATH_ENVAR = "ANSYSLIC_DIR"
+LIC_FILE_ENVAR = "ANSYSLMD_LICENSE_FILE"
+APP_NAME = "FEAT_ANSYS"  # TODO: We need to make sure this is the type of feature we need to checkout.
+LIC_NAME = 'meba' # TODO: We need to make sure this is the lest restrictive license.
+
+
+LOG = logging.getLogger(__name__)
+LOG.setLevel("CRITICAL")
+
+
+def check_license_file(timeout=10):
+    """Check the output of the license client log for connection error.
+
+    Expect type of errors with 'DENIED' in the header such as:
+    ```
+    2021/09/06 22:39:38    DENIED              ansys                           21.2 (2021.0512)             1/0/0/0                 1/1/1/1   10268:FEAT_ANSYS:gayuso@AAPDDqVK5WqNLrt.win.ansys.com:winx64   7368:192.168.18.10
+                Request name ansys does not exist in the licensing pool.
+                Cannot connect to license server system.
+                 The license server manager (lmgrd) has not been started yet,
+                 the wrong port@host or license file is being used, or the
+                 port or hostname in the license file has been changed.
+                Feature:       ansys
+                Server name:   192.168.18.10
+                License path:  1055@AAPDDqVK5WqNLrt;
+                FlexNet Licensing error:-15,578.  System Error: 10049 "WinSock: Invalid address"
+    ```
+
+    Parameters
+    ----------
+    timeout : int, optional
+        Time to keep checking the license log file for errors. Default to 10 (seconds).
```

Comment:
Perfect.  This is exactly what I need to be able to evaluate if this works and understand what it does.